### ISSUE TITLE
Does SW360 support Supports SAML authentication? 

### ISSUE THREAD

**Issue by KoukiHama** [eclipse/sw360#1332]
[//]: # (Copyright Siemens AG, 2021. Part of the SW360 Portal Project)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

### Description

I found some hints about SAML authentication by searching this rpeo
https://github.com/eclipse/sw360/search?q=SAML&type=code

But, I do not know well, how to realize SAML authentication in SW360/Liferay.

**Comment by mcjaeger** [comment]
yes, it is not SW360, but apache 2 and mod-mellon for apache 2. configuration is here:

https://github.com/sw360/sw360vagrant/blob/master/puppet/modules/sw360/templates/apache2-sw360.conf.erb

then you need to enable the login hook:

https://github.com/eclipse/sw360/blob/master/frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/users/SSOAutoLogin.java

by adding it here:

https://github.com/eclipse/sw360/blob/69175131d8afe87658102d82e44de86b645ac531/frontend/sw360-portlet/src/main/resources/sw360.properties#L118

**Comment by mcjaeger** [comment]
(note that the SSO login hook takes the mod-mellon header fields directly to perform the login. If that is done, it must be made sure that the liferay server cannot be contacted from elesewhere, only via apache2 forward, otherwise you can build simple requests performing auto login  ....
